encfs/encfssh

```diff
@@ -28,42 +29,63 @@ if [ ! -z "$2" ]; then
 	    mount_public=true
 	fi
     done
-    [ -d "$unenc_dir" ] || mkdir $unenc_dir
+    [ -d "$unenc_dir" ] || mkdir -- "$unenc_dir"
 else
     unenc_dir=$(mktemp -d .XXXXXXXX)
 fi
 
 if [ ! -d "$enc_dir" ]; then
-    mkdir $enc_dir
+    mkdir -- "$enc_dir"
 fi
 
 enc_dir=$(canonicalize "$enc_dir")
 unenc_dir=$(canonicalize "$unenc_dir")
 
-options=""
-if $unenc_dir_given; then
-    if $mount_public; then
-	options="-- -o allow_other"
+# clear command-line argument list; hereafter used for options
+set --
```

Comment:
It only *looks* simple -- the actual behavior of the options variant is considerably more complicated than how it reads. See [BashFAQ #50](http://mywiki.wooledge.org/BashFAQ/050) for an in-depth discussion of why strings are the wrong data structure to use for storing argument lists.